db/schema.rb

```diff
@@ -233,8 +233,10 @@
     t.datetime "file_updated_at",   null: false
     t.datetime "created_at",        null: false
     t.datetime "updated_at",        null: false
+    t.integer  "grant_id"
```

Comment:
user_type, user_id were intended to be whichever record "owns" this upload, even if not literally a user.  So instead of adding a grant_id, you can do `has_many :uploads, as: :user` in Grant model.